### PR TITLE
Configurable BERT layer reduction and lexer improvements

### PR DESCRIPTION
**Pull request by LoicGrobol**
This PR addresses #6, and brings with it some other improvements. As a hint of the influence of these changes, for Old French parsing, using mBERT and no pretrained FastText gives the following scores with the new settings

| Layers | Weigted | Frozen | UPOS_F | UAS_F  | LAS_F  | 
|--------|---------|--------|--------|--------|--------| 
| 4..8   | no      | no     | 93.83% | 88.49% | 82.56% | 
| 0..12  | no      | no     | 94.21% | 89.06% | 83.41% | 
| 0..12  | yes     | no     | 94.20% | 88.95% | 83.17% | 
| 0..12  | yes     | yes    | 94.53% | 89.37% | 83.75% | 

This is of course by no mean exhaustive testing, for which I think we should wait until #11 is addressed.

### Configs

- A new config option `bert_layers` that takes a list of integers to choose the layers to take into account when using a BERT model (defaults to the former behaviour `[4, 5, 6, 7]` when not provided, for backward compatibiliy).
- A new config option `bert_weighting` that takes a boolean. When it is false or absent, BERT word embeddings are averaged across the selected layers without weights (the former behaviour). When it is true, the average will be weighted with an ELMo-like scheme: if eᵢ is the embedding at the i-th layer, the final token embedding e will be e=γ∑ᵢwᵢeᵢ, with (wᵢ)=softmax(αᵢ), where γ and the αᵢ are trainable parameters of the lexer.
- A new config option `freeze_bert` that takes a boolean, which when true signals that the weights of the BERT model will not be trained and that it will be kept in eval state (e.g. without dropout) when we train the parser, thus only providing frozen representations (if using `bert_weighting`, the averaging weights of the lexer are still trained.

### Misc.

- A small utility function `lexers.freeze_module` that locks any `torch.nn.Module` in `eval` state for all intent and purposes. It lives in lexer for now since it's only used to freeze BERT models, but it might be useful at some other points.

## Changed

- Lexers now use the standard `train` and `eval` functions of `torch.nn.Module` instead of their custom `train_mode` and `eval_mode`, so they now automatically change their training state when the parent parser does.
- The ROOT token is now added as a special token for BERT tokenizers instead of a regular token (although we will probably have to revisit that while addressing #11)

### CODE REVIEW

**Comment by LoicGrobol** [comment]
Merging right away, since this does not break BC for the existing models and the CLI doesn't change. I don't think this should break anything as I have been testing it for several days now although I am not immune to mistakes :-)
